lib/gemonames.rb

```diff
@@ -36,17 +37,43 @@ def initialize(connection)
     end
 
     def search(query, country_code:, limit: 10)
-      perform_search_request(
-        query: query, country_code: country_code, max_rows: limit
-      ).fetch("geonames").map { |result|
+      perform_search :search,
+        query: query, country: country_code, maxRows: limit
+    end
+
+    def find(query, country_code:)
+      perform_find :search,
+        query: query, country: country_code
+    end
+
+    def reverse_search(latitude:, longitude:, limit: 10)
+      perform_search :find_nearby_place_name,
+        lat: latitude, lng: longitude, maxRows: limit
+    end
+
+    def reverse_find(latitude:, longitude:)
+      perform_find :find_nearby_place_name,
+        lat: latitude, lng: longitude
+    end
```

Comment:
I guess I should comment on the above changes:

I tried to achieve a shape where it is easily visible which endpoints are called on the geonames API and which parameters are passed exactly (that's why I changed to the ugly `maxRows` for example).